sstables/sstable_set.cc

```diff
@@ -75,6 +75,18 @@ class sstable_set_impl {
     virtual void insert(shared_sstable sst) = 0;
     virtual void erase(shared_sstable sst) = 0;
     virtual std::unique_ptr<incremental_selector_impl> make_incremental_selector() const = 0;
+
+    virtual flat_mutation_reader create_single_key_sstable_reader(
```

Comment:
If there is just a single "real" implementation... doesn't that make it "false abstraction"? Sorry couldn't resist :grin:.

Reply:
I think both implementations are "real": the default implementation (which uses `make_combined_reader`) and the twcs-specific implementation.

Reply:
Of course, I was just trolling :trollface:.

Reply:
And I started considering that nothing is real anymore :(